casper/src/test/scala/coop/rchain/casper/util/rholang/RuntimeManagerTest.scala

```diff
@@ -31,6 +34,48 @@ class RuntimeManagerTest extends FlatSpec with Matchers {
     result.status.isFailed should be(true)
   }
 
+  it should "capture rholang parsing errors and charge for parsing" in {
+    val badRholang = """ for(@x <- @"x"; @y <- @"y"){ @"xy"!(x + y) | @"x"!(1) | @"y"!("hi") """
+    val deploy     = ProtoUtil.sourceDeployNow(badRholang)
+    val (_, Seq(result)) =
+      runtimeManager
+        .use(mgr => mgr.computeState(mgr.emptyStateHash, deploy :: Nil))
+        .runSyncUnsafe(10.seconds)
+
+    result.status.isFailed should be(true)
+    result.cost.cost shouldEqual (accounting.parsingCost(badRholang).value)
```

Comment:
Strange... I'd expect this to read 'initial phlo available - parsingCost'

Reply:
This would give you the remaining phlo, not the real *cost* of the execution